code/modules/crafting/craft.dm

```diff
@@ -103,19 +103,25 @@
 	if(!R.tools.len)
 		return TRUE
 	var/list/possible_tools = list()
+	var/list/tools_used = list()
 	for(var/obj/item/I in user.contents)
 		if(istype(I, /obj/item/storage))
 			for(var/obj/item/SI in I.contents)
-				possible_tools += SI.type
-		possible_tools += I.type
-	possible_tools |= contents["other"]
-
+				if(SI.tool_behaviour) //Only add things that we could actually use as a tool
```

Comment:
@Fox-McCloud could you could take a look at this implementation of personal crafting that uses the new procs? It works fine but I wonder if there are better ways of doing it 😄

Reply:
looks good enough to me. I think we have other code that might be useful (the nuke disk checker) but this should(?) work fine